tests/regression/lp-1942266/task.yaml

```
@@ -0,0 +1,25 @@
+summary: Regression test for LP 1942266
+
+prepare: |
+  # build the first version
+  sed -e "s/##VERSION##/0.1/" ./test-system-files-conn-snap/meta/snap.yaml.in > ./test-system-files-conn-snap/meta/snap.yaml
+  sed -i -e "s/##EXTRA##//" ./test-system-files-conn-snap/meta/snap.yaml
+  snap pack ./test-system-files-conn-snap --filename=test-system-files-conn-snap-1.snap
+
+  # build the second version
+  sed -e "s/##VERSION##/0.2/" ./test-system-files-conn-snap/meta/snap.yaml.in > ./test-system-files-conn-snap/meta/snap.yaml
+  sed -i -e 's@##EXTRA##@- /etc/foo.conf @' ./test-system-files-conn-snap/meta/snap.yaml
+
+  snap pack ./test-system-files-conn-snap --filename=test-system-files-conn-snap-2.snap
+
+execute: |
+  echo "With the first revision, we don't have the rule because the interface is not connected"
+  snap install --dangerous test-system-files-conn-snap-1.snap
```

seems I forgot to add this but we should actually check that the rule isn't there to be double extra sure

```suggestion
  snap install --dangerous test-system-files-conn-snap-1.snap
  NOMATCH /etc/resolv.conf < /var/lib/snapd/apparmor/profiles/snap.test-system-files-conn-snap.bin
  NOMATCH /etc/foo.conf < /var/lib/snapd/apparmor/profiles/snap.test-system-files-conn-snap.bin
```